src/Parallel/spanningtrees/boruvka-multithread.jl

```diff
@@ -0,0 +1,123 @@
+using LightGraphs, SimpleWeightedGraphs
+using CPUTime
+using DelimitedFiles
+using Base
+using DataStructures
+using Base.Threads
+
+"""
+    boruvka_mst_multithread(g)
+
+Return a tuple `(mst, weights)` where `mst` is a vector of edges representing the
+optimum (minimum, by default) spanning tree of a connected, undirected graph
+`g` , and `weights` is the sum of all the edges in the solution by using
+[Boruvka's algorithm](https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm).
+The algorithm requires that all edges have different weights to correctly generate a minimun/maximum spanning tree
+### Optional parameter(s):
+`maxItr`: Used to limit maximum number of iterations the algorithm will be performed. The default is log2(numVertex).
+"""
+function boruvka_mst_multithread(g::SimpleWeightedGraph, maxItr = convert(Int64, round(log2(nv(g))+1, digits=0)))
+    connected_vs = IntDisjointSets(nv(g))
+    joined_nodes = Dict{Int, Vector{Int}}(i=>[i] for i in 1:nv(g))
+    MAX_WEIGHT=2000000000.0
+    minCost = fill(MAX_WEIGHT, nv(g)+1)
+    minNodeTgt = Vector{Int}(1:nv(g)+1)
+    minNodeSrc = Vector{Int}(1:nv(g)+1)
+    mst = Vector{edgetype(g)}()
+    sizehint!(mst, nv(g) - 1)
+    res =0
+    i=1
+    println("Max iteration: ", maxItr)
+    while(i<maxItr && length(mst)< nv(g)-1)
+        i+=1
+        initMinCostArray(g, minNodeSrc, minNodeTgt, minCost, MAX_WEIGHT)
+        findMinCostVertex(g, minNodeSrc, minNodeTgt, minCost, joined_nodes, connected_vs)
+        res+= contractVertex(g, minNodeSrc, minNodeTgt, minCost, joined_nodes, connected_vs,
+                mst, MAX_WEIGHT)
+    end
+
+    return (mst=mst, weight=res)
+end
+
+function initMinCostArray(
+        g::SimpleWeightedGraph,
+        minNodeSrc::Vector{Int},
+        minNodeTgt::Vector{Int},
+        minCost::Vector{Float64},
+        MAX_WEIGHT::Float64
+    )
+    for i = 1:nv(g)
+        minCost[i] = MAX_WEIGHT
+        minNodeTgt[i] = i
+        minNodeSrc[i] = i
+    end
+end
+
+function findMinCostVertex(
+        g::SimpleWeightedGraph,
+        minNodeSrc::Vector{Int},
+        minNodeTgt::Vector{Int},
+        minCost::Vector{Float64},
+        joined_nodes::Dict{Int, Vector{Int}},
+        connected_vs::IntDisjointSets,
+    )
+    sets = Vector{Int}(first.(keys(joined_nodes)))
```

Comment:
this is a bit misleading. `sets` is a Vector?

Reply:
Ah my bad. That variable was actually a list of roots from the disjoint sets, that's why I named it sets (and since all of the elements in the vectors themselves are unique.